extension/httpfs/httpfs.cpp

```diff
@@ -499,8 +500,25 @@ void HTTPFileHandle::Initialize(FileOpener *opener) {
 			length = 0;
 			return;
 		} else {
-			throw IOException("Unable to connect to URL \"" + path + "\": " + to_string(res->code) + " (" + res->error +
-			                  ")");
+			// HEAD request fail, use Range requet for another try (read only one byte)
+			if ((flags & FileFlags::FILE_FLAGS_READ)) {
+				char dummy_buffer[2];
```

Comment:
Can you add a comment explaining why this is necessary, or maybe it's better to edit `GetRangeRequest` to check for NULL and don't perform the memcpy in that case?